### PR TITLE
feat(anta.tests): PtpGMStatus, PtpLockStatus, PtpModeStatus, PtpOffset, PtpPortModeStatus

### PR DESCRIPTION
**Pull request by BarikiVarshitha**
# Description
Added test cases for PTP which include: PtpGMStatus, PtpLockStatus, PtpModeStatus, PtpOmfStatus, PtpPortModeStatus

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have run pre-commit for code linting and typing (`pre-commit run`)

### CODE REVIEW

**Comment by BarikiVarshitha** [comment]
Credits to [Gerard Phillips](mailto:gp@arista.com) for developing these test cases for PTP

File: anta/tests/ptp2.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+        time_to_last_sync = ptp_currentPtpSystemTime - ptp_lastSyncTime
+
+        if time_to_last_sync <= 60:
+            self.result.is_success(f"Current PTP lock found: '{time_to_last_sync}'s")
+        else:
+            self.result.is_failure(f"Device Lock is old: '{time_to_last_sync}'s")
+
+
+class PtpOmfStatus(AntaTest):
```
Comment (mtache) [inline anta/tests/ptp2.py:98]: I assume the test name is `PtpOfmStatus` and not `PtpOmfStatus` ?
What do you think about `PtpOffset` or `PtpOffsetStatus` ?
Not sure if this is relevant for some testing scenarios, but you could expose the 1000ns value as input with a default value.

**Review by mtache — CHANGES_REQUESTED** [review]
Thank you for your PR !
I have requested minor changes, please also write these tests in the `anta/tests/ptp.py` file that already exists and change the categories to `ptp` instead of `hardware`, that would make more sense.

File: anta/tests/ptp2.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+            self.result.is_success(f"Valid PTP mode found: '{ptp_mode}'")
+        else:
+            self.result.is_failure(f"Device is not configured as a Boundary Clock: '{ptp_mode}'")
+
+
+class PtpGMStatus(AntaTest):
+    """
+    This test verifies that the device is locked to a valid GM
+    The user should provide a single "validGM" as an input
+    To test PTP failover, re-run the test with secondary GMid configured.
```
[the remark this answers is not in the document]
Comment (BarikiVarshitha) [reply anta/tests/ptp2.py:40]: Do not want them to be hardcoded as they might be different for some other case, we give valid GM in the nrfu file so from there we are taking this value.

File: anta/tests/ptp2.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    """
+
+    class Input(AntaTest.Input):  # pylint: disable=missing-class-docstring
+        validGM: str
+        """ expected validGM as input from the user in nrfu file"""
```
[the remark this answers is not in the document]
Comment (BarikiVarshitha) [reply anta/tests/ptp2.py:45]: okay

File: anta/tests/ptp2.py [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    commands = [AntaCommand(command="show ptp", ofmt="json")]
+
+    # Verify that all switches are locked to the same GMID, and that this GMID is one of the provided GMs
+
+    @skip_on_platforms(["cEOSLab", "vEOS-lab"])
+    @AntaTest.anta_test
+    def test(self) -> None:
+        command_output = self.instance_commands[0].json_output
+        validGM = self.inputs.validGM
+        ptp_gmid = command_output["ptpClockSummary"]["gmClockIdentity"] if "gmClockIdentity" in command_output["ptpClockSummary"].keys() else ""
```
[the remark this answers is not in the document]
Comment (BarikiVarshitha) [reply anta/tests/ptp2.py:59]: okay

File: anta/tests/ptp2.py [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+    commands = [AntaCommand(command="show ptp", ofmt="json")]
+
+    # Verify that last lock time is within the last minute
+
+    @skip_on_platforms(["cEOSLab", "vEOS-lab"])
+    @AntaTest.anta_test
+    def test(self) -> None:
+        command_output = self.instance_commands[0].json_output
+        ptp_lastSyncTime = command_output["ptpClockSummary"]["lastSyncTime"] if "lastSyncTime" in command_output["ptpClockSummary"].keys() else ""
+        ptp_currentPtpSystemTime = (
```
[the remark this answers is not in the document]
Comment (BarikiVarshitha) [reply anta/tests/ptp2.py:85]: okay

File: anta/tests/ptp2.py [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+    categories = ["hardware"]
+    commands = [AntaCommand(command="show ptp", ofmt="json")]
+
+    # Verify that all switches are running Boundary Clock
+
+    @skip_on_platforms(["cEOSLab", "vEOS-lab"])
+    @AntaTest.anta_test
+    def test(self) -> None:
+        command_output = self.instance_commands[0].json_output
+        ptp_mode = command_output["ptpMode"] if "ptpMode" in command_output.keys() else ""
```
[the remark this answers is not in the document]
Comment (BarikiVarshitha) [reply anta/tests/ptp2.py:29]: okay

File: anta/tests/ptp2.py [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+    @AntaTest.anta_test
+    def test(self) -> None:
+        validPortModes = ["psMaster", "psSlave", "psPassive", "psDisabled"]
+        invalidPorts = 0
+        command_output = self.instance_commands[0].json_output
+        # print(command_output["ptpIntfSummaries"])
+
+        for interface in command_output["ptpIntfSummaries"]:
+            # print (interface)
+            # print(command_output["ptpIntfSummaries"][interface]["ptpIntfVlanSummaries"][0]['portState'])
```
[the remark this answers is not in the document]
Comment (BarikiVarshitha) [reply anta/tests/ptp2.py:152]: okay

File: anta/tests/ptp2.py [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+            for vlan in command_output["ptpIntfSummaries"][interface]["ptpIntfVlanSummaries"]:
+                # print(vlan['portState'])
```
[the remark this answers is not in the document]
Comment (BarikiVarshitha) [reply anta/tests/ptp2.py:154]: okay

File: anta/tests/ptp2.py [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+    commands = [AntaCommand(command="show ptp", ofmt="json")]
+
+    # Verify that ports are either Master / Slave / Passive or Disabled
+    @skip_on_platforms(["cEOSLab", "vEOS-lab"])
+    @AntaTest.anta_test
+    def test(self) -> None:
+        validPortModes = ["psMaster", "psSlave", "psPassive", "psDisabled"]
+        invalidPorts = 0
+        command_output = self.instance_commands[0].json_output
+        # print(command_output["ptpIntfSummaries"])
```
[the remark this answers is not in the document]
Comment (BarikiVarshitha) [reply anta/tests/ptp2.py:148]: okay

File: anta/tests/ptp2.py [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+
+        for interface in command_output["ptpIntfSummaries"]:
+            # print (interface)
+            # print(command_output["ptpIntfSummaries"][interface]["ptpIntfVlanSummaries"][0]['portState'])
+            for vlan in command_output["ptpIntfSummaries"][interface]["ptpIntfVlanSummaries"]:
+                # print(vlan['portState'])
+                if not vlan["portState"] in validPortModes:
+                    invalidPorts += 1
```
[the remark this answers is not in the document]
Comment (BarikiVarshitha) [reply anta/tests/ptp2.py:156]: used boolean

File: anta/tests/ptp2.py (hunk 1 above)
Comment (BarikiVarshitha) [reply anta/tests/ptp2.py:98]: using Ptpoffset

**Review by mtache — APPROVED** [review]
LGTM
Thank you for contributing !

**Pull request merged**